### ISSUE TITLE
Question on the Observation Encoder

### ISSUE THREAD

**Issue by JacobYuan7** [junwenchen/GAP_SRAM#1]
Hello author! It's so kind of you to make your codes public. And I raise several questions about your codes, which I hope you can help me resolve.
 
I suppose the **class st_gcn** in _st_gcn.py_ is the observation encoder. However, the forward function outputs **graph_boxes_features.mean(2).permute(0,2,1)**, which I think to be equal to **x.mean(2).permute(0,2,1)**. As I understand it, **x** is the input. Is this a bug?

**Comment by junwenchen** [comment]
Thanks for pointing out. We have updated it.

**Comment by JacobYuan7** [comment]
> Thanks for pointing out. We have updated it.

Thanks for updating! 
And I am trying to reproduce the results you reported in your paper, which I find hard to do so. There are several confusing points:
1. The optimizer you report in your paper is SGD, but you use Adam in this repo. Should I change it or keep with Adam?
2. The learning rate in your paper is 0:001, 0:001, 0:0001，but the learning rate starts from 0.00001 in your script, which is weird.
3. If I want to reproduce the results, shall I train a model with only the observation encoder, and then resume from it to train with the Activity Autoencoder and the Position Autoencoder? Should the observation encoder be frozen when training with the Activity Autoencoder and the Position Autoencoder?

Mind sharing any tips? Thanks!

**Comment by junwenchen** [comment]
> > Thanks for pointing out. We have updated it.
> 
> Thanks for updating!
> And I am trying to reproduce the results you reported in your paper, which I find hard to do so. There are several confusing points:
> 
> 1. The optimizer you report in your paper is SGD, but you use Adam in this repo. Should I change it or keep with Adam?
> 2. The learning rate in your paper is 0:001, 0:001, 0:0001，but the learning rate starts from 0.00001 in your script, which is weird.
> 3. If I want to reproduce the results, shall I train a model with only the observation encoder, and then resume from it to train with the Activity Autoencoder and the Position Autoencoder? Should the observation encoder be frozen when training with the Activity Autoencoder and the Position Autoencoder?
> 
> Mind sharing any tips? Thanks!

1. they both work
2. it should be 1e-4. The script was not updated. 
3. In this repo, I train them together, instead of the two stages.

**Comment by JacobYuan7** [comment]
> > > Thanks for pointing out. We have updated it.
> > 
> > 
> > Thanks for updating!
> > And I am trying to reproduce the results you reported in your paper, which I find hard to do so. There are several confusing points:
> > 
> > 1. The optimizer you report in your paper is SGD, but you use Adam in this repo. Should I change it or keep with Adam?
> > 2. The learning rate in your paper is 0:001, 0:001, 0:0001，but the learning rate starts from 0.00001 in your script, which is weird.
> > 3. If I want to reproduce the results, shall I train a model with only the observation encoder, and then resume from it to train with the Activity Autoencoder and the Position Autoencoder? Should the observation encoder be frozen when training with the Activity Autoencoder and the Position Autoencoder?
> > 
> > Mind sharing any tips? Thanks!
> 
> 1. they both work
> 2. it should be 1e-4. The script was not updated.
> 3. In this repo, I train them together, instead of the two stages.

Thanks for updating. I tried the second stage training on the whole dataset (not the **train_seqs** you define in _config.py_). It turns out that the result is around 88% for 100% partial observation. Have you tried this repo for reproducing the result you reported in your paper (91.97%)?
